### PR TITLE
Clarify information about artifacts and improve consistency in artifa…

### PR DESCRIPTION
**Pull request by discdiver**
Clarify information about artifacts and improve consistency in concepts->artifacts doc

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

File: docs/concepts/artifacts.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 # Artifacts
 
-Artifacts are persisted outputs such as tables, files, or links. They can be published via the Prefect SDK or API. They can also be rendered and managed in the Prefect UI, making it easy to track and monitor the objects that your flows produce and update over time. Published artifacts may be associated with a particular task run, flow run, or outside a flow run context. Artifacts provide a richer way to present information relative to typical logging practices-- including the ability to display tables, markdown, and links to external data.
+Artifacts are persisted outputs such as tables, files, or links. Artifacts are stored on the server can be published via the Prefect SDK or API. Artifacts be rendered and managed in the Prefect UI, making it easy to track and monitor the objects that your flows produce and update over time. Published artifacts may be associated with a particular task run, flow run, or outside a flow run context. Artifacts provide a richer way to present information relative to typical logging practices &mdash; including the ability to display tables, Markdown, and links to external data.
```
Comment (serinamarie) [inline docs/concepts/artifacts.md:3]: I don't think the first 3 sentences should start with "Artifacts" but I'll defer to @billpalombi

File: docs/concepts/artifacts.md [hunk 2]
```diff
@@ -230,6 +230,6 @@ for artifact in response.json():
     print(artifact)
 ```
 
-If we don't specify a key or that a key must exist, we will also return results (which are a type of key-less artifact).
+If you don't specify a key or that a key must exist, you will also return results (which are a type of key-less artifact).
```
Comment (serinamarie) [inline docs/concepts/artifacts.md:233]: thanks, I meant to correct the first-person plural

File: docs/concepts/artifacts.md (hunk 1 above)
Comment (billpalombi) [inline docs/concepts/artifacts.md:3]:
```suggestion
Artifacts are persisted outputs such as tables, files, or links. They can be published via the Prefect SDK or REST API. They are stored on Prefect server as well as rendered and managed in the Prefect UI, making it easy to track and monitor the objects that your flows produce and update over time. Published artifacts may be associated with a particular task run, flow run, or outside a flow run context. Artifacts provide a richer way to present information relative to typical logging practices &mdash; including the ability to display tables, Markdown, and links to external data.
```

**Review by billpalombi — COMMENTED** [review]
I agree with Serina's comments.

**Comment by discdiver** [comment]
Thanks! Updated.

**Review by billpalombi — APPROVED** [review]
lgtm

**Pull request merged**